R/http.R

```diff
@@ -23,30 +23,44 @@ httpRequest <- function(
   headers <- c(headers, authHeaders(authInfo, method, path), httpHeaders())
   certificate <- requestCertificate(service$protocol, authInfo$certificate)
 
-  # perform request
-  httpResponse <- httpLibCurl(
-    protocol = service$protocol,
-    host = service$host,
-    port = service$port,
-    method = method,
-    path = path,
-    headers = headers,
-    timeout = timeout,
-    certificate = certificate
-  )
-
-  while (isRedirect(httpResponse$status)) {
-    service <- redirectService(service, httpResponse$location)
+  if (isTRUE(getOption("rsconnect.httr2", TRUE))) {
```

Comment:
Should this start as disabled by default, merge, and then enable after additional testing?

Reply:
Related: The RStudio folks have some test automation for publishing workflows, but I think that runs against the CRAN release, not the GitHub version.

Reply:
Yeah that's what I was thinking, I'm happy to submit a PR to flip it off if it causes problems, but merging it TRUE might help us discover more quickly what's wrong (I'm sure there's something!).

I can kick off a CI run of Connect using rsconnect from this branch, see how that goes.